### ISSUE TITLE
CI testing Setup

### ISSUE THREAD

**Issue by DominiqueProulx** [Broke-Brokers/soen341_fproject2023#111]
From sprint 2 requirements :

> Set up your Continuous Integration pipeline in your repository. Unit tests must run automatically for every push/pull request to the main/master branch.

I think I found something. 
I enabled  the "action" tab in our repo, it was disabled by default. 

Here is the documentation 
https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-go

![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/5ecb6126-46fa-4049-a4f9-74299fb3e620)

**Comment by DominiqueProulx** [comment]
also : https://www.npmjs.com/package/testing

and :  
https://docs.cypress.io/guides/overview/why-cypress
from https://github.com/RGPosadas/Mull
This is from the Capstone project that is using Scrum and testing with cypress.

https://www.atlassian.com/continuous-delivery/continuous-integration

**Comment by DominiqueProulx** [comment]
From course notes.

![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/3b150086-3aa4-4eb0-9a32-0a2fd337a108)

![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/94e810d7-4491-4a7e-97ac-a64d57664d5b)
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/6b60da4e-8ba4-4822-8bb7-9af9077a86ab)

**Comment by DominiqueProulx** [comment]
TA is helping to set it up

We encountered errors in setting it up .

I successfully setup the github action test workflow 
![image](https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/assets/129322102/fbbcbcc0-1b85-4607-9e6c-6da85a6f1056)

**Comment by DominiqueProulx** [comment]
For creating the workflow : 

https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows

**Comment by DominiqueProulx** [comment]
I have finally been able to setup a test that runs a little bit , but doe snot complete

Here is the workflow  : https://github.com/Broke-Brokers/Broke_BRokers-soen341projectF2023/blob/main-branch/.github/workflows/IntegrationTest_1.js.yml 

I have added every dependency 

It bugs at : tests.  Because we have not setup any tests yet

I have added a App.test.js file to the src. and added a smoke test from https://create-react-app.dev/docs/running-tests.
